docs/API.md

```diff
@@ -0,0 +1,96 @@
+# API
+
+Before going deeply in the Tartiflette API, don't forget to take a look of the [getting started](./getting-started.md) section.
+
+## Engine initialization
+
+The way to generate the engine is pretty simple.
+
+The engine accepts at least one parameter, called "sdl".
+
+```python
+tartiflette.Engine(sdl)
+```
+
+**When the `sdl` parameters contains the raw schema.**
+
+```python
+import tartiflette
+
+engine = tartiflette.Engine(
+    """
+    type Query {
+        hello: String
+    }
+    """
+)
+```
+
+**When the `sdl` parameters target a file.**
+
+The filepath specified have to contain the full schema definition language.
+
+```python
+import tartiflette
+
+engine = tartiflette.Engine(
+    "/User/chuck/workspace/mytartiflette/schema.sdl"
+)
+```
+
+**When the `sdl` parameters target a file list.**
+
+Every files will be concatenated, one after the other.
+
+```python
+import tartiflette
+
+engine = tartiflette.Engine(
+    [
+        "/User/chuck/workspace/mytartiflette/schema_query.sdl",
+        "/User/chuck/workspace/mytartiflette/schema_mutation.sdl"
+    ]
+)
+```
+
+**When the `sdl` parameters target a folder.**
+
+Every files which are suffixed by `.sdl` will be concatenated, one after the other.
+
+```python
+import tartiflette
+
+engine = tartiflette.Engine(
+    "/User/chuck/workspace/mytartiflette"
+)
+```
+
+### Advanced constructor
+
+The engine provided an advanced interface to be initialized. It accepts optionals and named parameters.
+
+```python
+tartiflette.Engine(
+    sdl,
+    resolver_middlewares=[],
+    resolvers={},
+    directive_resolvers={},
+    guess_resolvers_from_decorator=True,
+)
+```
+
+1. **sdl:** Schema Definition Language, detailed above.
+2. **resolver_middlewares:** Middlewares list which are applied **only** on resolvers.
+3. **resolvers:** By default, the resolvers list is guessed from the `@Resolver` decorator. Is it possible to pass the list of resolver
+```
+def my_hello_resolver(parent, args, ctx, info):
+    return "Hey"
+
+resolvers = {
+    "Query": {
+        "hello": my_hello_resolver
+    }
+}
+```
+4. **directive_resolvers:** Resolver list which are applied on the directives.
+5. **guess_resolvers_from_decorator:** Enable the automatic resolution of Resolver through the built-in Tartiflette decorator, e.g. `@tartiflette.Resolver('Query.Hello)`.
```

Comment:
This is not guess work so the term should be something else ... 

Enable the automatic link between `@Resolver` decorators and the SDL.